cornflow/shared/authentication.py

```diff
@@ -76,10 +80,30 @@ def get_user_obj_from_header(headers):
         data = Auth.decode_token(token)
         user_id = data["user_id"]
         user = UserModel.get_one_user(user_id)
-        if not user:
+        if user is None:
             raise ObjectDoesNotExist("User does not exist, invalid token")
         return user
 
+    @staticmethod
+    def get_permission_for_request(req, user_id):
+        method, url = Auth.get_request_info(req)
+        user_role = UserRoleModel.get_one_user(user_id=user_id)
+        if user_role is None:
+            raise NoPermission("You do not have permission to access this endpoint")
+
+        action_id = PERMISSION_METHOD_MAP[method]
+        view_id = ApiViewModel.query.filter_by(url_rule=url).first().id
+
+        for role in user_role:
```

Comment:
alternatively, couldn't you send the list of roles to the query and make only one query that intersects the list with the roles that have permission?

Reply:
I will see how to do it with sending the ROLES_WITH_ACCESS parameter of the endpoint.